INFINITY-2974: Update binary secrets documentation.

mpereira opened the pull request.

nickbp commented on docs/pages/ops-guide/overview.md:
```diff
@@ -266,19 +266,19 @@ Configuring `ROOT` vs `MOUNT` volumes may depend on the service. Some services w
 ## Virtual networks
 
-The SDK allows pods to join virtual networks, with the `dcos` virtual network available by defualt. You can specify that a pod should join the virtual network by using the `networks` keyword in your YAML definition. Refer to [Developers Guide](developer-guide.md) for more information about how to define virtual networks in your service.
+The SDK allows pods to join virtual networks, with the `dcos` virtual network available by defualt. You can specify that a pod should join the virtual network by using the `networks` keyword in your YAML definition. Refer to [Developers Guide](../developer-guide.md) for more information about how to define virtual networks in your service.
```
FWIW these are about to conflict with related changes here: https://github.com/mesosphere/dcos-commons/pull/2211/files#diff-e39adaadbdc093952734c3c8e4d19c4eR269

benclarkwood replied:
Does it make sense, @nickbp, for you to just pull these tweaks into your changes?

nickbp replied:
The other PR is switching to rendering pages in dedicated directories for pretty urls (i.e., instead of `/developer-guide.html`, it's `/developer-guide/index.html`), in order to align with what docs.mesosphere.com does. So the correct links themselves are changing as of that PR

mpereira replied:
@nickbp I'll let your PR merge first and resolve the conflicts here.

mpereira replied:
Done: https://github.com/mesosphere/dcos-commons/pull/2213/commits/9a2e08bc97f11d9338a153264ab1a1d5f5bf5117

elezar commented on docs/pages/ops-guide/overview.md:
````diff
@@ -310,27 +310,37 @@ The path of a secret defines which service IDs can have access to it. You can th
+To create a secret called `mysecret` with the binary contents of `kerb5.keytab` run:
 
-You can use standard `base64` command line utility. The following example uses the BSD `base64` command.
 ```
````
`bash` tag.

elezar commented on docs/pages/ops-guide/overview.md:
````diff
@@ -310,27 +310,37 @@ The path of a secret defines which service IDs can have access to it. You can th
 
 ```
-$  dcos security secrets  create -f kerb5.keytab.base64-encoded  some/path/__dcos_base64__mysecret
+$ base64 -w 0 -i krb5.keytab > kerb5.keytab.base64-encoded
````
The `-i` flag here is not the same as `--input` in the MacOS case:
```
  -i, --ignore-garbage  when decoding, ignore non-alphabet characters
```

elezar commented on docs/pages/ops-guide/overview.md:
````diff
@@ -310,27 +310,37 @@ The path of a secret defines which service IDs can have access to it. You can th
+
 ```
-$  base64 -w 0 -i krb5.keytab > kerb5.keytab.base64-encoded
+$ base64 -i krb5.keytab -o kerb5.keytab.base64-encoded
````
We could specify the long options `--input` and `--output` instead.

Also `bash` tag.

elezar commented on docs/pages/ops-guide/overview.md:
````diff
@@ -310,27 +310,37 @@ The path of a secret defines which service IDs can have access to it. You can th
-only as a file such that it will be autoatically decoded and made available as a temporary in-memory file mounted within your container (file-based secrets).
+Now that the file is encoded it can be stored as a secret. There's one important detail: the secret name **must** be prefixed with `__dcos_base64__`. For example `some/path/__dcos_base64__mysecret` and `__dcos_base64__mysecret` will be base64-decoded automatically by DC/OS.
+
+```
````
`bash` tag.

elezar commented on docs/pages/ops-guide/overview.md:
```diff
@@ -310,27 +310,37 @@ The path of a secret defines which service IDs can have access to it. You can th
 
-When you reference the `__dcos_base64__mysecret` secret in your service, the content of the secret will be first base64-decoded, and then copied and made available to your service. Refer to the [Developer Guide](developer-guide.md) for more information on how to reference DC/OS secrets as a file in SDK-based services. Refer to a binary secret
-only as a file such that it will be autoatically decoded and made available as a temporary in-memory file mounted within your container (file-based secrets).
+Now that the file is encoded it can be stored as a secret. There's one important detail: the secret name **must** be prefixed with `__dcos_base64__`. For example `some/path/__dcos_base64__mysecret` and `__dcos_base64__mysecret` will be base64-decoded automatically by DC/OS.
```
In other section we typically have a `*Note:*` block.

I would move the discussion on the name requirements to just after the last paragraph below as a `*Note:*`.

mpereira commented:
Thanks @elezar. I added the bash tags https://github.com/mesosphere/dcos-commons/pull/2213/commits/4e59b724b8cc3b175e19e7cc287d32c6cc08322d and moved that section to a note block (with some rewording to make the last paragraph easier to understand) https://github.com/mesosphere/dcos-commons/pull/2213/commits/695b7f6e4e158c07da31ca7264b504c9e344d6d1

Let me know what you think.
